File: client/src/type/filesystem.ts
export type IFileType = "directory" | "file" | "link" | "char";
export interface IFilesystem {
	type: IFileType;
	name: string;
	path: string;
	mode: string;
	prot: string;
	size: number;
	time: number;
	contents: IFilesystem[] | null;
}

File: server/pkg/model/filesystem.go
package model

type Filesystem struct {
	Type string `json:"type"`
	Name string `json:"name"`
	User string `json:"user"`
	Size int    `json:"size"`
	Mode string `json:"mode"`
	Prot string `json:"prot"`
	Time string `json:"time"`

	Contents []Filesystem `json:"contents"`
}
